setup.cfg

```
@@ -28,10 +27,10 @@ project_urls =
     Documentation = https://dandi.readthedocs.io
 
 [options]
-python_requires = >=3.7
+python_requires = >=3.8
 install_requires =
     appdirs
-    bidsschematools >= 0.5.0
+    bidsschematools >= 0.7.0
```

Should `< 0.8.0` be made an upper bound so we don't have to make another emergency PR like #1293?

---

+1 on that , so 
```suggestion
    bidsschematools ~= 0.7.0
```
?

---

Wasn't really an emergency PR:

(1) it existed for over a month
(2) the mistake was me not [limiting the version](https://github.com/dandi/dandi-cli/pull/1293) when I made it so that this could have been handled without temporarily breaking CI.

Adding artificial upper bounds I think is more dangerous, since we might end up forgetting to update.
In the case of BST, which is just me in another repo, I think either is fine.

---

I was referring to #1293 (which you created today) as the "emergency PR," and if you say that the mistake was not limiting the version, isn't that an argument in support of limiting the version going forwards?

---

not really, since we don't know for a fact that there will be any API-breaking changes in 0.8, or 0.9, or 0.10 etc. On the contrary I actually suspect there won't. If I start suspecting there will, yes, I will remember to add the limit this time :3

---

Honestly It's fine if you really want this....

Just that there's an epidemic of people artificially setting upper bounds and they're almost always pointless and needlessly confuse package management.